### PR TITLE
feat: more adlib-action functions

### PR DESCRIPTION
**Pull request by Julusian**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This will let adlib-actions do more things.
The aim here is to let the actions change the transition on the next-part

This needs some unit tests and some more manual testing before being ready to merge

* **Other information**:

**Status**

- [x] The functionality has been tested by the PR author
- [ ] The functionality has been tested by NRK

### CODE REVIEW

File: meteor/server/api/blueprints/__tests__/context-adlibActions.test.ts [hunk 1]
```diff
@@ -130,11 +84,73 @@ describe('Test blueprint api context', () => {
 		})
 	}
 
+	function generateSparsePieceInstances2(rundown: Rundown) {
```
Comment (nytamin) [inline meteor/server/api/blueprints/__tests__/context-adlibActions.test.ts:87]: 2? is there a 1? (and a 0?) :)

File: meteor/server/api/blueprints/context/adlibActions.ts [hunk 2]
```diff
@@ -431,6 +387,57 @@ export class ActionExecutionContext extends ShowStyleContext implements IActionE
 
 		return clone(unprotectObject(newPartInstance))
 	}
+	updatePartInstance(part: 'current' | 'next', props: Partial<IBlueprintMutatablePart>): IBlueprintPartInstance {
```
Comment (nytamin) [inline meteor/server/api/blueprints/context/adlibActions.ts:390]: me & @Julusian talked about this one the other day, I noted down that initially we won't allow editing the current part due to having to handle side-effects, but we'll keep `part: 'next'` for future compatibility.

File: meteor/server/api/blueprints/__tests__/context-adlibActions.test.ts (hunk 1 above)
Comment (Julusian) [reply meteor/server/api/blueprints/__tests__/context-adlibActions.test.ts:87]: Ah, yeah there is a `generateSparsePieceInstances` also, I meant to refactor the tests to be consistent but forgot. Ill tidy that all up

File: meteor/server/api/blueprints/context/adlibActions.ts (hunk 2 above)
Comment (Julusian) [reply meteor/server/api/blueprints/context/adlibActions.ts:390]: I did that for removePieceInstances, but not for updatePartInstance. 
I think that keeping this one to edit the current is reasonable, and matches the flexibility given with updatePieceInstance, and is generally safer/less destructive than removePieceInstances

File: meteor/server/api/blueprints/__tests__/context-adlibActions.test.ts (hunk 1 above)
Comment (nytamin) [reply meteor/server/api/blueprints/__tests__/context-adlibActions.test.ts]: never mind me, I got things mixed up.

File: meteor/server/api/blueprints/context/adlibActions.ts (hunk 2 above)
Comment (nytamin) [reply meteor/server/api/blueprints/context/adlibActions.ts:390]: never mind me, I got things mixed up here as well..

**Pull request merged**